app/uk/gov/hmrc/merchandiseinbaggage/repositories/DeclarationRepository.scala

```diff
@@ -49,26 +51,29 @@ trait DeclarationRepository {
 }
 
 @Singleton
-class DeclarationRepositoryImpl @Inject()(mongo: () => DB)(implicit ec: ExecutionContext)
-    extends ReactiveRepository[Declaration, String]("declaration", mongo, Declaration.format, implicitly[Format[String]])
-    with DeclarationDateOrdering with DeclarationRepository {
-
-  implicit val jsObjectWriter: OWrites[JsObject] = new OWrites[JsObject] {
-    override def writes(o: JsObject): JsObject = o
-  }
-
-  override def indexes: Seq[Index] = Seq(
-    Index(Seq(s"${Declaration.id}" -> Ascending), Option("primaryKey"), unique = true)
-  )
+class DeclarationRepositoryImpl @Inject()(mongo: MongoComponent)(implicit ec: ExecutionContext)
```

Comment:
The old code injected had a function that returned a DB, ie () => DB

The new code injects MongoComponent rather than a function that reutrns one, e.g. () => MongoComponent

The function allows late binding, do we still need to keep that rather than direct injection?